client/lib/fifo/fifo_windows.go

```diff
@@ -78,13 +78,17 @@ func New(path string) (io.ReadWriteCloser, error) {
 		return nil, err
 	}
 
-	return &winFIFO{
-		listener: l,
-	}, nil
+	openFn := func() (io.ReadCloser, error) {
+		return &winFIFO{
+			listener: l,
+		}, nil
+	}
+
+	return openFn, nil
 }
 
-// OpenWriter opens a fifo that already exists and returns an io.ReadWriteCloser for it
-func Open(path string) (io.ReadWriteCloser, error) {
+// OpenWriter opens a fifo that already exists and returns an io.WriteCloser for it
+func Open(path string) (io.WriteCloser, error) {
 	return winio.DialPipe(path, nil)
 }
 
```

Comment:
Q: Do we haveexisting tests that exercise fifo on windows codepath

Reply:
I have enabled fifo package tests in windows, and current fifo_test.go files pass on Windows too:  https://ci.appveyor.com/project/hashicorp/nomad/builds/23511423 .

FWIW, I have tried logmon tests but I was getting some errors related to pipe name; I may look into it in another PR.